pkg/destinations/redshift/query.go

```diff
@@ -0,0 +1,143 @@
+package redshift
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/rs/zerolog/log"
+)
+
+func (s *RedshiftServer) QueryJSON(query string, writer io.Writer) error {
+	rows, err := s.conn.Query(query)
+	if err != nil {
+		log.Err(err).Msg("failed to execute query")
+		return err
+	}
+	defer rows.Close()
+
+	columns, err := rows.Columns()
+	if err != nil {
+		log.Err(err).Msg("failed to get column names")
+		return err
+	}
+
+	values := make([]interface{}, len(columns))
+	valuePtrs := make([]interface{}, len(columns))
+	for i := range columns {
+		valuePtrs[i] = &values[i]
+	}
+
+	_, err = writer.Write([]byte("["))
+	if err != nil {
+		log.Err(err).Msg("failed to write JSON array start:")
+		return err
+	}
+
+	firstRow := true
+	for rows.Next() {
+		err := rows.Scan(valuePtrs...)
+		if err != nil {
+			log.Err(err).Msg("failed to scan row values")
+			return err
+		}
+
+		jsonObject := make(map[string]interface{})
+		for i, column := range columns {
+			jsonObject[column] = values[i]
+		}
+
+		jsonData, err := json.Marshal(jsonObject)
```

Comment:
Would it make sense to use json.Encoder instead? https://pkg.go.dev/encoding/json#NewEncoder